src/posts/2021-01-20-ebpf-updates/index.md

```diff
@@ -0,0 +1,315 @@
+  environments in terms of networking, security, and observability.
+
+<!--2021-01-12-->
+* [Kubernetes Podcast - _Episode #133: Cilium, with Thomas Graf_](https://kubernetespodcast.com/episode/133-cilium/),
+  from Craig Box and Adam Glick.  
+  Interviewed by Craig and Adam, Thomas Graf recounts the advent of eBPF, and
+  how it introduced a new paradigm for network processing at a time when the
+  other firewalling solutions would derive their behavior from hardware and
+  would fail to scale. The episode goes on with explanations on how Cilium and
+  Hubble were built on top of eBPF to provide network policies and monitoring
+  for clusters. More advanced questions on the relationship with Envoy, or on a
+  possible support for eBPF on Windows, follow. “Twenty-two years on, do you
+  think you finally fixed networking?” Future will tell.
+
+<!--2021-01-18-->
+* Optimyze.cloud [announced in a tweet](https://twitter.com/OptimyzeCloud/status/1351203905850519557)
+  that they are working on a “_Full-system lightweight continuous profiling for
+  Linux Kernel, C/C++, Rust, Golang, Python, JVM, PHP (with Ruby and Node
+  planned for the future)_”, apparently based on eBPF.
+
+<!--2021-01-18-->
+* [_Securing Containerized Environments with eBPF_](https://www.tag-cyber.com/advisory/articles/securing-containerized-environments-with-ebpf),
+  from Adam LeWinter.  
+  Following the transition from physical hardware to virtual machines, most
+  workflows are now moving to containers. In this context, new challenges in
+  terms of visibility and security arise, Adam explains. Cilium leverages eBPF
+  to provide network routing and observability. It combines metadata from
+  layers 3 and 4 with layer 7 parameters such as the HTTP method, bringing
+  “_visibility and enforcement based on a service, pod, or container
+  identity_”.
+
+### Software Projects
+
+* ipftrace2, a tool to track packets inside the Linux kernel,
+  [got a new v0.1.0 release](https://github.com/YutaroHayakawa/ipftrace2/releases/tag/v0.1.0),
+  bringing new features such as support for writing extension module with C
+  thanks to CO-RE, thus improving programmability and portability.
+
+### Software: Sandbox and Experiments
+
+* [tc-skeleton](https://github.com/florianl/tc-skeleton) is a simple example
+  project to demonstrate how to load eBPF programs with
+  [go-tc](https://github.com/florianl/go-tc), a work in progress version of tc
+  (the Linux tool for traffic control) written in Go.
+
+* [lnetd-host-encap](https://github.com/cpmarvin/lnetd-ctl/tree/main/host-encap)
+  is an experiment where an eBPF program encapsulate packets with MPLS headers.
+
+## The Kernel Side
+
+Below are some highlights for the first
+[pull request for the bpf-next tree](https://lore.kernel.org/bpf/20210116012922.17823-1-daniel@iogearbox.net/t/#u)
+for the 5.12 cycle:
+
+* Add atomic operations to eBPF. To that goal, extend the eBPF instruction set
```

Comment:
`With that goal`/`To that end`, no?
```suggestion
* Add atomic operations to eBPF. To that end, extend the eBPF instruction set
```

Reply:
Was looking for that one, it wouldn't come back to my brain! :pray: